lib/mayaUsd/ufe/UsdTransform3dPointInstance.cpp

```diff
@@ -0,0 +1,270 @@
+        _position = GfVec3f(transform.GetTranslation());
+        _orientation = GfQuath(transform.GetRotation().GetQuat());
+        _scale = GfVec3f(transform.GetScale());
+    }
+
+    ~UsdPointInstanceSetMatrix4dUndoableCmd() override = default;
+
+    bool set(const Ufe::Matrix4d&) override
+    {
+        // No-op: Maya does not set matrices through interactive manipulation.
+        TF_WARN("Illegal call to UsdPointInstanceSetMatrix4dUndoableCmd::set()");
+        return true;
+    }
+
+    void executeUndoBlock() override
+    {
+        UsdSceneItem::Ptr usdItem = downcast(sceneItem());
+
+        if (!usdItem || !usdItem->isPointInstance()) {
+            return;
+        }
+
+        const UsdPrim& usdPrim = usdItem->prim();
+        const int      instanceIndex = usdItem->instanceIndex();
+        _positionModifier.setPrimAndInstanceIndex(usdPrim, instanceIndex);
+        _orientationModifier.setPrimAndInstanceIndex(usdPrim, instanceIndex);
+        _scaleModifier.setPrimAndInstanceIndex(usdPrim, instanceIndex);
+
+        _positionModifier.setValue(_position, UsdTimeCode::Default());
+        _orientationModifier.setValue(_orientation, UsdTimeCode::Default());
+        _scaleModifier.setValue(_scale, UsdTimeCode::Default());
+    }
+
+private:
+    GfVec3f                             _position;
+    GfQuath                             _orientation;
+    GfVec3f                             _scale;
+    UsdPointInstancePositionModifier    _positionModifier;
+    UsdPointInstanceOrientationModifier _orientationModifier;
+    UsdPointInstanceScaleModifier       _scaleModifier;
+};
+
+} // anonymous namespace
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+UsdTransform3dPointInstance::UsdTransform3dPointInstance(const UsdSceneItem::Ptr& item)
+    : UsdTransform3dBase(item)
+{
+    if (item != nullptr) {
+        const UsdPrim prim = item->prim();
+        const int     instanceIndex = item->instanceIndex();
+        _positionModifier.setPrimAndInstanceIndex(prim, instanceIndex);
+        _orientationModifier.setPrimAndInstanceIndex(prim, instanceIndex);
+        _scaleModifier.setPrimAndInstanceIndex(prim, instanceIndex);
+    }
+}
+
+/* static */
+UsdTransform3dPointInstance::Ptr UsdTransform3dPointInstance::create(const UsdSceneItem::Ptr& item)
+{
+    return std::make_shared<UsdTransform3dPointInstance>(item);
+}
+
+/* override */
+Ufe::Vector3d UsdTransform3dPointInstance::translation() const
+{
+    return _positionModifier.getUfeValue();
+}
+
+/* override */
+Ufe::Vector3d UsdTransform3dPointInstance::rotation() const
+{
+    return _orientationModifier.getUfeValue();
+}
+
+/* override */
+Ufe::Vector3d UsdTransform3dPointInstance::scale() const { return _scaleModifier.getUfeValue(); }
+
+/* override */
+Ufe::TranslateUndoableCommand::Ptr
+UsdTransform3dPointInstance::translateCmd(double x, double y, double z)
+{
+    return std::make_shared<UsdPointInstanceTranslateUndoableCommand>(
+#ifdef UFE_V2_FEATURES_AVAILABLE
```

Comment:
You added UsdTransform3dPointInstance.cpp to the UFE_V2 part of the CMakeLists.txt file, so either that's incorrect, or you can remove the UFE_V2_FEATURES_AVAILABLE ifdefs here.

Reply:
Yup, thanks for this. As noted above, I think the former is likely incorrect and `UsdTransform3dPointInstance` should get compiled in for both versions of UFE.

Reply:
As it turns out, `UsdTransform3dPointInstance` inherits from `UsdTransform3dBase`, which only gets compiled for UFE v2, so I updated the CMakeLists.txt to only build in `UsdTransform3dPointInstance` for UFE v2 as well.